packages/ihe-gateway/server/Channels/CareQuality Document Exchange - Outbound/XCA ITI-39 Processor/destinationConnector-XCA Endpoint-responseTransformer-step-1-Case 1 - Success or PartialSuccess.js

```diff
@@ -27,79 +27,92 @@ if ('Success' == queryResponseCode.toString() || 'PartialSuccess' == queryRespon
 		// Process retrieved documents
 		for each (var entry in xml.*::DocumentResponse) {
 
-			var attachment = {};
-			attachment.homeCommunityId = entry.*::HomeCommunityId.toString().replace('urn:oid:', '');
-			attachment.repositoryUniqueId = entry.*::RepositoryUniqueId.toString().replace('urn:uuid:', '');
-			attachment.docUniqueId = entry.*::DocumentUniqueId.toString().replace('urn:uuid:', '');
-			attachment.metriportId = idMapping[attachment.docUniqueId.toString()];
-			attachment.fileLocation = bucketName;
-
-			// Responding Gateways which support the Persistence of Retrieved Documents Option shall specify the NewRepositoryUniqueId element
-			// indicating the document is available for later retrieval and be able to return exactly the same document in all future retrieve
-			// requests for the document identified by NewDocumentUniqueId.
-			var newRepositoryUniqueId = entry.*::NewRepositoryUniqueId.toString();
-			if (newRepositoryUniqueId) attachment.newRepositoryUniqueId = newRepositoryUniqueId.toString();
+			try {
+				var attachment = {};
+				attachment.homeCommunityId = entry.*::HomeCommunityId.toString().replace('urn:oid:', '');
+				attachment.repositoryUniqueId = entry.*::RepositoryUniqueId.toString().replace('urn:uuid:', '');
+				attachment.docUniqueId = entry.*::DocumentUniqueId.toString().replace('urn:uuid:', '');
+				attachment.metriportId = idMapping[attachment.docUniqueId.toString()];
+				attachment.fileLocation = bucketName;
+
+				const logError = (ex) => {
+					logger.error("Error decoding doc - docUniqueId: " + attachment.docUniqueId + "; metriportId: " +
+						attachment.metriportId + "; " + ex);
+				};
 
-			var newDocumentUniqueId = entry.*::NewDocumentUniqueId.toString();
-			if (newDocumentUniqueId) attachment.newDocumentUniqueId = newDocumentUniqueId.toString();
+				// Responding Gateways which support the Persistence of Retrieved Documents Option shall specify the NewRepositoryUniqueId element
+				// indicating the document is available for later retrieval and be able to return exactly the same document in all future retrieve
+				// requests for the document identified by NewDocumentUniqueId.
+				var newRepositoryUniqueId = entry.*::NewRepositoryUniqueId.toString();
+				if (newRepositoryUniqueId) attachment.newRepositoryUniqueId = newRepositoryUniqueId.toString();
 
-			// Files are stored in format: <CX_ID>/<PATIENT_ID>/<CX_ID>_<PATIENT_ID>_<DOC_ID>
+				var newDocumentUniqueId = entry.*::NewDocumentUniqueId.toString();
+				if (newDocumentUniqueId) attachment.newDocumentUniqueId = newDocumentUniqueId.toString();
 
-			try {
+				let decodedAsString = null;
+				try {
+					var documentEncodedString = entry.*::Document.toString();
+					var decoded = FileUtil.decode(documentEncodedString);
+					decodedAsString = new Packages.java.lang.String(decoded);
+				} catch (ex) {
+					logError(ex);
+				}
+				if (!decodedAsString) continue;
```

Comment:
OOC - when would we expect this to every throw an error or be null? Only if the string isn't valid b64 I presume?

Reply:
I'm not sure, but trying to be defensive here so if that happens we keep processing the remaining documents.